zcash_primitives/src/sighash_versioning.rs

```diff
@@ -0,0 +1,112 @@
+//! Sighash versioning as specified in [ZIP-246].
+//!
+//! [ZIP-246]: https://zips.z.cash/zip-0246
+
+use alloc::{collections::BTreeMap, vec::Vec};
+use lazy_static::lazy_static;
+
+use orchard::{
+    issuance_sighash_versioning::IssueSighashVersion,
+    orchard_sighash_versioning::OrchardSighashVersion,
+};
+
+/// The sighash version and associated data
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SighashInfo {
+    version: u8,
+    associated_data: Vec<u8>,
+}
+
+impl SighashInfo {
+    /// Constructs a `SighashInfo` from raw bytes.
+    ///
+    /// Returns `None` if `bytes` is empty.
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        bytes.split_first().map(|(&version, info)| Self {
+            version,
+            associated_data: info.to_vec(),
+        })
+    }
```

Comment:
probably better to
```rust

impl TryFrom<&[u8]> for SighashInfo {
    type Error = ();
    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
        let (&version, rest) = bytes.split_first().ok_or(())?;
        Ok(SighashInfo { version, associated_data: rest.to_vec() })
    }
}

Reply:
I removed SighashInfo struct and its impl since it is no longer used (after removing ORCHARD/ISSUE_SIGHASH_VERSION_TO_INFO)
So, this function no longer exists